### ISSUE TITLE
Build problem

### ISSUE THREAD

**Issue by blakemcbride** [lem-project/lem#1129]
I am running make and getting:

```
$ make
qlot install
which: no ros in (/home/blake/bin:/home/blake/bin:/home/blake/.local/bin:/home/blake/bin:/usr/local/bin:/usr/bin:/bin:/home/blake/.local/share/JetBrains/Toolbox/scripts:/usr/local/sbin:/usr/sbin:/home/blake/.dotnet/tools:/opt/plan9port/bin:/opt/icon/bin:/opt/groovy-4.0.12/bin:/home/blake/.local/share/JetBrains/Toolbox/scripts:/opt/plan9port/bin:/opt/icon/bin:/opt/groovy-4.0.12/bin)
Reading '/home/blake/Backup/lem.git/qlfile'...
Already have dist "quicklisp" version "2023-06-18".
Already have dist "micros" version "git-ed264a27262baeed493cdde338873bf4afc3721c".
Already have dist "lem-mailbox" version "git-12d629541da440fadf771b0225a051ae65fa342a".
Already have dist "lem-base16-themes" version "git-07dacae6c1807beaeffc730063b54487d5c82eb0".
Already have dist "async-process" version "git-9690530fc92b59636d9f17d821afa7697e7c8ca4".    
Already have dist "sblint" version "git-1037296f604c3210ce073a53539d4ae95b0c2f8c".       
Already have dist "rove" version "git-f168cd177b5f83f171dd970dc1a9abb6eb43f044".  
Already have dist "cl-sdl2" version "git-24dd7f238f99065b0ae35266b71cce7783e89fa7".
Already have dist "cl-sdl2-ttf" version "git-e61bb2119003d8ae7792d38aa11f7728d3ee5a00".
Already have dist "cl-sdl2-image" version "git-8734b0e24de9ca390c9f763d9d7cd501546d17d4".
Loading '/home/blake/Backup/lem.git/contrib/bracket-paren-mode/lem-bracket-paren-mode.asd'...
Loading '/home/blake/Backup/lem.git/contrib/calc-mode/lem-calc-mode.asd'...                  
...
Loading '/home/blake/Backup/lem.git/lib/process/lem-process.asd'...  
Loading '/home/blake/Backup/lem.git/lib/socket-utils/lem-socket-utils.asd'...
Loading '/home/blake/Backup/lem.git/src/base/lem-base.asd'...                
Ensuring 88 dependencies installed.
Unhandled END-OF-FILE in thread #<SB-THREAD:THREAD tid=49735 "main thread" RUNNINGux-threads-v0.9.1.tgz".
                                   {10010D0003}>:
  end of file on #<SB-SYS:FD-STREAM for "descriptor 6" {1001168013}>

Backtrace for: #<SB-THREAD:THREAD tid=49735 "main thread" RUNNING {10010D0003}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<END-OF-FILE {1003042C23}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<END-OF-FILE {1003042C23}>)
2: (INVOKE-DEBUGGER #<END-OF-FILE {1003042C23}>)
3: (ERROR END-OF-FILE :STREAM #<SB-SYS:FD-STREAM for "descriptor 6" {1001168013}>)
4: (READ-LINE #<SB-SYS:FD-STREAM for "descriptor 6" {1001168013}> T NIL #<unused argument>)
5: (QLOT/SECURE-DOWNLOADER::HTTPS-FETCH "http://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz" #P"/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz")
6: ((:METHOD QL-DIST:ENSURE-LOCAL-ARCHIVE-FILE (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
7: ((:METHOD QL-DIST:INSTALL (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
8: ((:METHOD QL-DIST:ENSURE-INSTALLED (T)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
9: (QLOT/INSTALL::INSTALL-DEPENDENCIES #P"/home/blake/Backup/lem.git/" #P"/home/blake/Backup/lem.git/.qlot/")
10: (QLOT/INSTALL:INSTALL-QLFILE #P"/home/blake/Backup/lem.git/qlfile" :QUICKLISP-HOME NIL :INSTALL-DEPS T :CACHE-DIRECTORY NIL)
11: (QLOT/CLI::QLOT-COMMAND "install")
12: (QLOT/CLI:MAIN)
13: (SB-INT:SIMPLE-EVAL-IN-LEXENV (QLOT/CLI:MAIN) #<NULL-LEXENV>)
14: (EVAL (QLOT/CLI:MAIN))
15: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:LOAD . "/home/blake/.qlot/qlot/.qlot/setup.lisp") (:EVAL . "(asdf:load-asd #P\"/home/blake/.qlot/qlot/qlot.asd\")") (:EVAL . "(ql:quickload :qlot/cli :silent t)") (:EVAL . "(qlot/cli:main)") (:QUIT)))
16: (SB-IMPL::TOPLEVEL-INIT)
17: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
18: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
19: (SB-IMPL::%START-LISP)

unhandled condition in --disable-debugger mode, quitting
make: *** [Makefile:4: ncurses] Error 1
$
```

**Comment by cxxxr** [comment]
Is your qlot version up to date?
Also, @fukamachi might have better advice.

**Comment by blakemcbride** [comment]
I downloaded qlot afresh.

**Comment by blakemcbride** [comment]
I think life would be easier for all by using MAKE or a shell script rather than yet another dependency.

**Comment by cxxxr** [comment]
Indeed, it would be nice if we could get the complicated procedure into a makefile or something...

**Comment by cxxxr** [comment]
By the way, did you solve this issue?

**Comment by fukamachi** [comment]
It seems to be a network problem with S3.
Do you have the same problem constantly?
If so, tell me your OS, your Lisp, and those versions.

**Comment by blakemcbride** [comment]
I am on:
    64 bit Fedora Linux 38
    SBCL 2.3.8

I just tried again with a current pull.  Same issue.

Thanks!

**Comment by fukamachi** [comment]
Can you try it with the latest Qlot? It should show more detailed logs for its downloader.

**Comment by blakemcbride** [comment]
When I type: qlot --version
I get: 1.2.10

I do not know how to upgrade it.  What version should I have?

Thanks!!

**Comment by blakemcbride** [comment]
I tried "pip install ros" and I get an error "Python.h: No such file or directory".

I think the problem is that it builds for you because, over time, you have installed all of the prerequisites.  You're just assuming that I am starting from the same point (same environment).  I am not a python programmer, and in spite of the fact that lisp is my favorite language, I get no professional opportunity to use lisp.

What I suggest is this.  Create a clean VM.  Figure out and document the steps to build lem.

**Comment by blakemcbride** [comment]
Python --version returns 3.11.6

My research seems to indicate that Python 3.11.6 is too recent.  Others with my error say that Python 3.10 works.  Fedora uses the latest releases.

**Comment by blakemcbride** [comment]
Just FYI.  I am doing a presentation on Lisp next month.  I had hoped to show lem, too.

**Comment by blakemcbride** [comment]
I am the author of [github.com/blakemcbride/Kiss](https://github.com/blakemcbride/Kiss)
It had a complex build that was causing me endless build issues.  I tried ant, maven, and gradle.  It was a nightmare.
I ended up writing my own build system that can easily handle any level of complexity.  It is part of that Kiss package.
It is also used in a few commercial packages.
If it interests you, I'd be happy to re-write it in lisp.  That would give you a lisp-only build solution that can easily handle any build complexities.

**Comment by mateodif** [comment]
I experienced this issue while using my quicklisp installation of qlot, which seems to be outdated (it was version 1.0.1). 
After installing qlot using the "automatic installer", it now builds just fine. FYI, at the time of writing this comment, latest qlot version is 1.2.13.

**Comment by fukamachi** [comment]
@blakemcbride
Though `which` shows an error message unexpectedly, Qlot is fine without Roswell if `sbcl` is installed.
You can upgrade it with just the same way to install it: `curl -L https://qlot.tech/installer | bash`.

**Comment by blakemcbride** [comment]
After updating qlot:
```
$ sbcl --version
SBCL 2.3.8
$ qlot --version
Qlot 1.2.13
```
Clean clone gives:
```
$ make
qlot install
Installing Quicklisp to /home/blake/Backup/lem.git/.qlot/...
Reading '/home/blake/Backup/lem.git/qlfile'...              
Already have dist "quicklisp" version "2023-06-18".
Installing dist "micros".                          
=> Newly installed "micros" version "git-ed264a27262baeed493cdde338873bf4afc3721c".
Installing dist "lem-mailbox".                                                     
=> Newly installed "lem-mailbox" version "git-12d629541da440fadf771b0225a051ae65fa342a".
Installing dist "lem-base16-themes".                                                    
=> Newly installed "lem-base16-themes" version "git-07dacae6c1807beaeffc730063b54487d5c82eb0".
Installing dist "async-process".                                                              
=> Newly installed "async-process" version "git-9690530fc92b59636d9f17d821afa7697e7c8ca4".
Installing dist "sblint".                                                                 
=> Newly installed "sblint" version "git-1037296f604c3210ce073a53539d4ae95b0c2f8c".
Installing dist "rove".                                                            
=> Newly installed "rove" version "git-f168cd177b5f83f171dd970dc1a9abb6eb43f044".
Installing dist "cl-sdl2".                                                       
=> Newly installed "cl-sdl2" version "git-24dd7f238f99065b0ae35266b71cce7783e89fa7".
Installing dist "cl-sdl2-ttf".                                                      
=> Newly installed "cl-sdl2-ttf" version "git-e61bb2119003d8ae7792d38aa11f7728d3ee5a00".
Installing dist "cl-sdl2-image".                                                        
=> Newly installed "cl-sdl2-image" version "git-8734b0e24de9ca390c9f763d9d7cd501546d17d4".
Loading '/home/blake/Backup/lem.git/contrib/bracket-paren-mode/lem-bracket-paren-mode.asd'...
Loading '/home/blake/Backup/lem.git/contrib/calc-mode/lem-calc-mode.asd'...                  
Loading '/home/blake/Backup/lem.git/contrib/fbar/lem-fbar.asd'...          
Loading '/home/blake/Backup/lem.git/contrib/google-translate/lem-google-translate.asd'...
[...]
Loading '/home/blake/Backup/lem.git/lib/process/lem-process.asd'...  
Loading '/home/blake/Backup/lem.git/lib/socket-utils/lem-socket-utils.asd'...
Loading '/home/blake/Backup/lem.git/src/base/lem-base.asd'...                
Ensuring 88 dependencies installed.
Unhandled QL-DIST:MISSING-LOCAL-ARCHIVE in thread #<SB-THREAD:THREAD tid=130047 "main thread" RUNNING0.9.1.tgz".
                                                     {10010D0003}>:
  The archive file "bordeaux-threads-v0.9.1.tgz" for release "bordeaux-threads" is missing

Backtrace for: #<SB-THREAD:THREAD tid=130047 "main thread" RUNNING {10010D0003}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<QL-DIST:MISSING-LOCAL-ARCHIVE {1002AC32A3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<QL-DIST:MISSING-LOCAL-ARCHIVE {1002AC32A3}>)
2: (INVOKE-DEBUGGER #<QL-DIST:MISSING-LOCAL-ARCHIVE {1002AC32A3}>)
3: (ERROR QL-DIST:MISSING-LOCAL-ARCHIVE :FILE #P"/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz" :RELEASE #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>)
4: ((:METHOD QL-DIST:CHECK-LOCAL-ARCHIVE-FILE (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
5: ((:METHOD QL-DIST:ENSURE-LOCAL-ARCHIVE-FILE (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
6: ((:METHOD QL-DIST:INSTALL (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
7: ((:METHOD QL-DIST:ENSURE-INSTALLED (T)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
8: (QLOT/INSTALL::INSTALL-DEPENDENCIES #P"/home/blake/Backup/lem.git/" #P"/home/blake/Backup/lem.git/.qlot/")
9: (QLOT/INSTALL:INSTALL-QLFILE #P"/home/blake/Backup/lem.git/qlfile" :QUICKLISP-HOME NIL :INSTALL-DEPS T :CACHE-DIRECTORY NIL)
10: (QLOT/CLI::QLOT-COMMAND "install")
11: (QLOT/CLI:MAIN)
12: (SB-INT:SIMPLE-EVAL-IN-LEXENV (QLOT/CLI:MAIN) #<NULL-LEXENV>)
13: (EVAL (QLOT/CLI:MAIN))
14: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:LOAD . "/home/blake/.qlot/qlot/.qlot/setup.lisp") (:EVAL . "(asdf:load-asd #P\"/home/blake/.qlot/qlot/qlot.asd\")") (:EVAL . "(ql:quickload :qlot/cli :silent t)") (:EVAL . "(qlot/cli:main)") (:QUIT)))
15: (SB-IMPL::TOPLEVEL-INIT)
16: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
17: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
18: (SB-IMPL::%START-LISP)

unhandled condition in --disable-debugger mode, quitting
make: *** [Makefile:4: ncurses] Error 1
$ 
````

**Comment by fukamachi** [comment]
Unfortunately, I couldn't reproduce it on a Docker container of Fedora 38, which has `sbcl`, `curl`, `which`, `make` and `git`.

**Comment by fukamachi** [comment]
Can you run the following shell command in Lem's directory?

```bash
sbcl --noinform --no-userinit --no-sysinit --non-interactive --load .qlot/setup.lisp --eval '(ql:quickload :lem-ncurses :verbose t)'
```

**Comment by blakemcbride** [comment]
I did a make, got the same error.  Then I ran:
```
$ sbcl --noinform --no-userinit --no-sysinit --non-interactive --load .qlot/setup.lisp --eval '(ql:quickload :lem-ncurses :verbose t)'
To load "lem-ncurses":
  Load 1 ASDF system:
    lem-ncurses
; Loading "lem-ncurses"
..............To load "bordeaux-threads":
  Load 5 ASDF systems:
    alexandria asdf global-vars trivial-features
    trivial-garbage
  Install 1 Quicklisp release:
    bordeaux-threads
; Fetching 'https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz'.
####################################################################Unhandled CL+SSL::SSL-ERROR-SSL in thread #<SB-THREAD:THREAD tid=336514 "main thread" RUNNING
                                             {10044900A3}>:
  A failure in the SSL library occurred on handle #.(SB-SYS:INT-SAP #X5616D09DFD80) (SSL_get_error: 1). ERR_print_errors(): 40479F80687F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:ssl/record/rec_layer_s3.c:303:


Backtrace for: #<SB-THREAD:THREAD tid=336514 "main thread" RUNNING {10044900A3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<CL+SSL::SSL-ERROR-SSL {100172EBD3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<CL+SSL::SSL-ERROR-SSL {100172EBD3}>)
2: (INVOKE-DEBUGGER #<CL+SSL::SSL-ERROR-SSL {100172EBD3}>)
3: (ERROR CL+SSL::SSL-ERROR-SSL :HANDLE #.(SB-SYS:INT-SAP #X5616D09DFD80) :RET 1 :PRINTED-QUEUE "40479F80687F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:ssl/record/rec_layer_s3.c:303:
" :QUEUE NIL)
4: (CL+SSL::SSL-SIGNAL-ERROR #.(SB-SYS:INT-SAP #X5616D09DFD80) #<FUNCTION CL+SSL::SSL-READ> 1 0)
5: ((:METHOD TRIVIAL-GRAY-STREAMS:STREAM-READ-SEQUENCE (CL+SSL::SSL-STREAM T T T)) #<unavailable argument> #<unavailable argument> #<unavailable argument> #<unavailable argument>) [fast-method]
6: ((:METHOD STREAM-READ-SEQUENCE (TRIVIAL-GRAY-STREAMS:FUNDAMENTAL-INPUT-STREAM T)) #<CL+SSL::SSL-STREAM for #<FD-STREAM for "socket 192.168.4.34:39662, peer: 18.161.84.41:443" {10053E7C63}>> #(231 26 0 230 204 240 236 194 129 28 25 108 ...) 0 NIL) [fast-method]
7: (READ-SEQUENCE #(231 26 0 230 204 240 236 194 129 28 25 108 ...) #<CL+SSL::SSL-STREAM for #<FD-STREAM for "socket 192.168.4.34:39662, peer: 18.161.84.41:443" {10053E7C63}>> :START 0 :END NIL)
8: (QLOT/FETCH::FETCH-FILE "https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz" "/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz" :QUIETLY NIL)
9: (SB-INT:SIMPLE-EVAL-IN-LEXENV (QLOT/FETCH::MAIN) #<NULL-LEXENV>)
10: (EVAL (QLOT/FETCH::MAIN))
11: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:LOAD . "/home/blake/.qlot/qlot/.qlot/setup.lisp") (:EVAL . "(asdf:load-asd #P\"/home/blake/.qlot/qlot/qlot.asd\")") (:EVAL . "(ql:quickload :qlot/fetch :silent t)") (:EVAL . "(qlot/fetch::main)") (:QUIT)))
12: (SB-IMPL::TOPLEVEL-INIT)
13: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
15: (SB-IMPL::%START-LISP)

unhandled condition in --disable-debugger mode, quitting
Unhandled UIOP/RUN-PROGRAM:SUBPROCESS-ERROR in thread #<SB-THREAD:THREAD tid=336512 "main thread" RUNNING
                                                         {10044900A3}>:
  Subprocess #<UIOP/LAUNCH-PROGRAM::PROCESS-INFO {10039213D3}>
 with command ("/home/blake/.qlot/qlot/scripts/fetch.sh"
               "https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz"
               "/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz")
 exited with error code 1

Backtrace for: #<SB-THREAD:THREAD tid=336512 "main thread" RUNNING {10044900A3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<UIOP/RUN-PROGRAM:SUBPROCESS-ERROR {100392B133}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<UIOP/RUN-PROGRAM:SUBPROCESS-ERROR {100392B133}>)
2: (INVOKE-DEBUGGER #<UIOP/RUN-PROGRAM:SUBPROCESS-ERROR {100392B133}>)
3: (CERROR "IGNORE-ERROR-STATUS" UIOP/RUN-PROGRAM:SUBPROCESS-ERROR :COMMAND ("/home/blake/.qlot/qlot/scripts/fetch.sh" "https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz" "/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz") :CODE 1 :PROCESS #<UIOP/LAUNCH-PROGRAM::PROCESS-INFO {10039213D3}>)
4: (UIOP/RUN-PROGRAM::%CHECK-RESULT 1 :COMMAND ("/home/blake/.qlot/qlot/scripts/fetch.sh" "https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz" "/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz") :PROCESS #<UIOP/LAUNCH-PROGRAM::PROCESS-INFO {10039213D3}> :IGNORE-ERROR-STATUS NIL)
5: (UIOP/RUN-PROGRAM::%USE-LAUNCH-PROGRAM ("/home/blake/.qlot/qlot/scripts/fetch.sh" "https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz" "/home/blake/Backup/lem.git/.qlot/dists/quicklisp/archives/bordeaux-threads-v0.9.1.tgz") :OUTPUT :INTERACTIVE :ERROR-OUTPUT :INTERACTIVE)
6: ((:METHOD QL-DIST:ENSURE-LOCAL-ARCHIVE-FILE (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
7: ((:METHOD QL-DIST:INSTALL (QL-DIST:RELEASE)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
8: ((:METHOD QL-DIST:ENSURE-INSTALLED (T)) #<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>) [fast-method]
9: (SB-KERNEL:%MAP-FOR-EFFECT-ARITY-1 QL-DIST:ENSURE-INSTALLED (#<QL-DIST:RELEASE bordeaux-threads-v0.9.1 / quicklisp 2023-06-18>))
10: (QUICKLISP-CLIENT::APPLY-LOAD-STRATEGY #<QUICKLISP-CLIENT::LOAD-STRATEGY "bordeaux-threads" (5 asdf, 1 quicklisp)>)
11: (QUICKLISP-CLIENT::AUTOLOAD-SYSTEM-AND-DEPENDENCIES "bordeaux-threads" :PROMPT NIL)
12: (QUICKLISP-CLIENT::AUTOLOAD-SYSTEM-AND-DEPENDENCIES "lem-ncurses" :PROMPT NIL)
13: ((:METHOD QUICKLISP-CLIENT:QUICKLOAD (T)) :LEM-NCURSES :PROMPT NIL :SILENT NIL :VERBOSE T) [fast-method]
14: (QL-DIST::CALL-WITH-CONSISTENT-DISTS #<FUNCTION (LAMBDA NIL :IN QUICKLISP-CLIENT:QUICKLOAD) {100334383B}>)
15: (SB-INT:SIMPLE-EVAL-IN-LEXENV (QUICKLISP-CLIENT:QUICKLOAD :LEM-NCURSES :VERBOSE T) #<NULL-LEXENV>)
16: (EVAL (QUICKLISP-CLIENT:QUICKLOAD :LEM-NCURSES :VERBOSE T))
17: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:LOAD . ".qlot/setup.lisp") (:EVAL . "(ql:quickload :lem-ncurses :verbose t)") (:QUIT)))
18: (SB-IMPL::TOPLEVEL-INIT)
19: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
20: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
21: (SB-IMPL::%START-LISP)

unhandled condition in --disable-debugger mode, quitting
$ 
```

**Comment by blakemcbride** [comment]
Interestingly, I can download https://beta.quicklisp.org/archive/bordeaux-threads/2023-06-18/bordeaux-threads-v0.9.1.tgz via my browser.

**Comment by fukamachi** [comment]
It could be an issue in CL+SSL, but I can't tell any further since I can't reproduce it on my machine.
Can I have the version of your libssl?

**Comment by blakemcbride** [comment]
Although I use SSH, SCP, and HTTPS every day, I do not really see libssl except for what vmware uses.  (I use VMWare Workstation for other stuff.)

```
$ cd /lib
$ find . -name 'libssl*'
./vmware/lib/libssl.so.3
./vmware/lib/libssl.so.3/libssl.so.3
./vmware-ovftool/libssl.so.1.0.2
./vmware-vix/Workstation-17.0.0/64bit/libssl.so.3
$ cd /usr/lib
$ find . -name 'libssl*'
./vmware/lib/libssl.so.3
./vmware/lib/libssl.so.3/libssl.so.3
./vmware-ovftool/libssl.so.1.0.2
./vmware-vix/Workstation-17.0.0/64bit/libssl.so.3
$ cd /usr/local/lib
$ find . -name 'libssl*'
$ 
```

**Comment by blakemcbride** [comment]
I tried to build lem on a brand new Fedora 38 system and got the same error.  Fedora has a habit of installing the latest versions of things.

**Comment by blakemcbride** [comment]
I tried a brand new LinuxMint and got the same error.

**Comment by blakemcbride** [comment]
I tried disabling my firewall.  Same error.

**Comment by blakemcbride** [comment]
Have you tried creating a new VM and building using your own instructions?

**Comment by Sasanidas** [comment]
If you are in Linux Mint, would you mind to try the automatic installer? (https://lem-project.github.io/lem-page/installation/ncurses/linux/#debianmintubuntu)

**Comment by blakemcbride** [comment]
My guess is it'll work.  Getting a working version is not what I'm after.  I'm after the ability to build it.  Thanks!

**Comment by Sasanidas** [comment]
Oh I see! But in this case, it should be similar (so if you can have that version, you should be able to build it)

**Comment by blakemcbride** [comment]
Yes.  I should be able to build it.  I followed the build instructions but it failed as I described.  Here is what I think we're dealing with:

1.  The system was built with old instructions that worked.  Part of the build process involved some needed steps.

2. The build instructions changed.  It works for people who had built in the past because they previously included a step that is no longer present in the instructions.

Therefore, new people cannot build the system.  I followed the instructions several times on several different new systems and got the same error each time.  Until someone who knows the system well tries to build lem on a new system and corrects the instructions, new people will be unable to build the system and your growth will be stunted.

I did a presentation to a large group and was unable to demonstrate lem because of this.  I have stopped using lem because of this.  I'll try it again in a year or so.
